common/src/main/java/io/netty/util/NetUtil.java

```diff
@@ -863,7 +795,9 @@ public static Inet6Address getByName(CharSequence ip, boolean ipv4Mapped) {
                         (ipv6Separators == IPV6_MAX_SEPARATORS &&
                           (compressBegin <= 2 && ip.charAt(0) != ':' ||
                            compressBegin >= 14 && ip.charAt(tmp) != ':'))) ||
-                    currentIndex + 1 >= bytes.length) {
+                    currentIndex + 1 >= bytes.length ||
+                    begin < 0 && (ip.charAt(tmp) != ':' || ip.charAt(tmp - 1) != ':') ||
```

Comment:
was this a bug, and if so can you please isolate out this change and the corresponding unit test into a separate PR?